### PR TITLE
Refactor in public ip resource and datasource

### PR DESCRIPTION
**Pull request by coderGo93**
- Removed value `Standard` for validation of parameter `Sku` because is not supported.
-  Removed parameter `public_ip_address_allocation` in favor of parameter `allocation_method`
-  Removed value IPv6 for validation because is not supported
-  Removed tests that involve Standard SKU or IPv6
- Updated docs for resource and datasource of Public IP

### CODE REVIEW

File: internal/services/network/public_ip_resource.go [hunk 1]
```diff
@@ -54,24 +53,9 @@ func publicIp() *pluginsdk.Resource {
...
-				Optional:         true,
-				Computed:         true,
-				DiffSuppressFunc: suppress.CaseDifference,
-				ExactlyOneOf:     []string{"public_ip_address_allocation", "allocation_method"},
-				Deprecated:       "the `public_ip_address_allocation` property has been renamed to `allocation_method`",
-				ValidateFunc: validation.StringInSlice([]string{
-					string(network.Dynamic),
-					string(network.Static),
-				}, true),
-			},
```
Comment (katbyte) [inline internal/services/network/public_ip_resource.go:68]: was this not supported int 0.10 of the provider? we can't just remove a property like that without first deprecating it

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/network/public_ip_resource.go (hunk 1 above)
Comment (katbyte) [reply internal/services/network/public_ip_resource.go:68]: if we are going to make hard breaking changes like this in 1.0 of the provider we need to give it some care and make sure existing users are prepared and ok with it

Comment (coderGo93) [reply internal/services/network/public_ip_resource.go]: Done, thanks for the comment

Comment (katbyte) [reply internal/services/network/public_ip_resource.go]: we will have to deprecate this in a 0.11 release, but that can be handled prior to relerase 1.0

**Review by katbyte — APPROVED** [review]
LGTM

**Pull request merged**